x-pack/filebeat/module/iptables/log/ingest/pipeline.json

```diff
@@ -0,0 +1,179 @@
+                    "ignore_missing": true
+                }
+            },
+            {
+                "geoip": {
+                    "field": "source.ip",
+                    "target_field": "source.geo",
+                    "ignore_missing": true
+                }
+            },
+            {
+                "geoip": {
+                    "field": "destination.ip",
+                    "target_field": "destination.geo",
+                    "ignore_missing": true
+                }
+            },
+            {
+                "script": {
+                    "lang": "painless",
+                    "params": {
+                        "mappings": [
+                            {
+                                "source": {
+                                    "object": "iptables",
+                                    "key": "ether_type"
+                                },
+                                "destination": {
+                                    "object": "network",
+                                    "key": "type"
+                                },
+                                "map": {
+                                    "08:00": "ipv4",
+                                    "86:dd": "ipv6"
+                                }
+                            },
+                            {
+                                "source": {
+                                    "object": "event",
+                                    "key": "outcome"
+                                },
+                                "destination": {
+                                    "object": "event",
+                                    "key": "outcome"
+                                },
+                                "map": {
+                                    "D": "deny",
+                                    "A": "allow"
+                                }
+                            },
+                            {
+                                "source": {
+                                    "object": "network",
+                                    "key": "transport"
+                                },
+                                "destination": {
+                                    "object": "network",
+                                    "key": "transport"
+                                },
+                                "map": {
+                                    "icmpv6": "ipv6-icmp"
+                                }
+                            }
+                        ]
+                    },
+                    "source": "for (action in params.mappings) { def src = ctx[action.source.object]; if (src != null) { Map map = action.map; String key = src[action.source.key]; String mapping = map[key]; if (mapping != null) { Map dst = ctx[action.destination.object]; if (dst == null) { dst = new HashMap(); ctx[action.destination.object] = dst;} dst[action.destination.key] = mapping; } } }"
```

Comment:
Ah, I see it now. Thanks for explaining.

Only thing I see with this is that right now you're limited to a depth of 2 fields (e.g. `network.transport`) for source and destination field names. But I'm not sure it's worth modifying for this detail, as we don't need to support deeper nesting for now.

Check out how much I could pack in a screenful [in 10192](https://github.com/elastic/beats/pull/10192/files#diff-9875b63218295fb1abdd7d4556c9a34cR92), though. Have you considered just using the basic processors and formatting them more aggressively? ;-)

I still see a need for some custom code for the mapping of values, so just something to think about. No action needed.

By the way, I've been wanting something similar to this in Ingest Node in general. Since right now is a bit late to make requests for 7.0 (understatement of the day), I'm making a note of your code. We could potentially extract it out to a reusable pipeline, pack a bunch of mappings in a temp variable, and call this shared "mapping pipeline". I've noted down this idea in issue https://github.com/elastic/ecs/issues/181